pv211_utils/beir/loader.py

```diff
@@ -0,0 +1,294 @@
+            raw_train_data = combine_beir_datasets(raw_train_data, temp_train_data)
+        if dataset.dev:
+            temp_dev_data = load_beir_dev_set(dataset.name, data_path, dataset.dev_alternative)
+            raw_dev_data = combine_beir_datasets(raw_dev_data, temp_dev_data)
+        if dataset.test:
+            temp_test_data = load_beir_test_set(dataset.name, data_path, dataset.test_alternative)
+            raw_test_data = combine_beir_datasets(raw_test_data, temp_test_data)
+    return raw_train_data, raw_dev_data, raw_test_data
+
+
+def load_queries(raw_queries, max_test_queries: Optional[int] = None, query_class=BeirQueryBase) -> OrderedDict:
+    queries = OrderedDict()
+    if max_test_queries is None:
+        max_test_queries = 99967454511  # the entire Big Smoke's order is close enough to infinity
+    for que in list(raw_queries.items())[:max_test_queries]:
+        que_id = que[0]
+        query = query_class(
+            query_id=que_id,
+            body=que[1],
+        )
+        queries[que_id] = query
+    return queries
+
+
+def load_documents(raw_corpus, document_class=BeirDocumentBase) -> OrderedDict:
+    from collections import OrderedDict
+    documents = OrderedDict()
+    for doc in raw_corpus.items():
+        doc_id = doc[0]
+        document = document_class(
+            document_id=doc_id,
+            body=doc[1]["text"]
+        )
+        documents[doc_id] = document
+    return documents
+
+
+def load_judgements(queries: OrderedDict, raw_documents: OrderedDict, raw_qrels) -> BeirJudgementsBase:
+    judgements = set()
+    for que_id in queries.keys():
+
+        for doc_id in (list(raw_qrels[que_id].keys())):
+            query: BeirQueryBase = queries[que_id]
+            document: BeirDocumentBase = raw_documents[doc_id]
+            relevance = (query, document)
+            judgements.add(relevance)
+    return judgements
+
+
+def create_false_judgements(documents: OrderedDict, queries: OrderedDict, judgements: BeirJudgementsBase,
+                            num_of_false_judgements: int = 10000) -> BeirJudgementsBase:
+
+    false_judgements = set()
+    seed = 42
+    randomizer = random.Random(seed)
+    random.seed(42)
+
+    doc_ids = list(documents.keys())
+    que_ids = list(queries.keys())
+
+    for _ in range(num_of_false_judgements):
+        que_id = randomizer.choice(que_ids)
+        doc_id = randomizer.choice(doc_ids)
+        query: BeirQueryBase = queries[que_id]
+        document: BeirDocumentBase = documents[doc_id]
+        relevance = (query, document)
+
+        while relevance in judgements or relevance in false_judgements:
+            que_id = randomizer.choice(que_ids)
+            doc_id = randomizer.choice(doc_ids)
+            query: BeirQueryBase = queries[que_id]
+            document: BeirDocumentBase = documents[doc_id]
+            relevance = (query, document)
+        false_judgements.add(relevance)
+    return false_judgements
```

Comment:
What is the purpose of this code? The `create_false_judgements()` function seems unused in the code base. If this is a helper function for a user who needs negative sampling, it should be documented so that its purpose is clear for the user.

Reply:
todo

Reply:
should be done

Reply:
> should be done

What has changed?